mypyc/emitwrapper.py

```diff
@@ -42,6 +42,8 @@ def generate_wrapper_function(fn: FuncIR, emitter: Emitter) -> None:
     optional_pos = [arg for arg in real_args if arg.kind == ARG_OPT]
     optional_kwonly = [arg for arg in real_args if arg.kind == ARG_NAMED_OPT]
     required_kwonly = [arg for arg in real_args if arg.kind == ARG_NAMED]
+    star_args = next((arg for arg in real_args if arg.kind == ARG_STAR), None)
```

Comment:
Some of the cleanups stuff was able to be pretty cleanly moved into `generate_wrapper_core`, which ends up being nicer, since now it can take care of figuring out what the right thing to do on an error is.

I also moved format string generation out, facilitated by grouping arguments into a list instead of separate variables.